src/Microsoft.Extensions.DependencyModel/Resolution/PackageCompilationAssemblyResolver.cs

```diff
@@ -11,74 +11,95 @@ namespace Microsoft.Extensions.DependencyModel.Resolution
     public class PackageCompilationAssemblyResolver: ICompilationAssemblyResolver
     {
         private readonly IFileSystem _fileSystem;
-        private readonly string _nugetPackageDirectory;
+        private readonly string[] _nugetPackageDirectories;
 
         public PackageCompilationAssemblyResolver()
             : this(EnvironmentWrapper.Default, FileSystemWrapper.Default)
         {
         }
 
         public PackageCompilationAssemblyResolver(string nugetPackageDirectory)
-            : this(FileSystemWrapper.Default, nugetPackageDirectory)
+            : this(FileSystemWrapper.Default, new string[] {nugetPackageDirectory})
         {
         }
 
         internal PackageCompilationAssemblyResolver(IEnvironment environment,
             IFileSystem fileSystem)
-            : this(fileSystem, GetDefaultPackageDirectory(environment))
+            : this(fileSystem, GetDefaultProbeDirectories(environment))
         {
         }
 
-        internal PackageCompilationAssemblyResolver(IFileSystem fileSystem, string nugetPackageDirectory)
+        internal PackageCompilationAssemblyResolver(IFileSystem fileSystem, string[] nugetPackageDirectories)
         {
             _fileSystem = fileSystem;
-            _nugetPackageDirectory = nugetPackageDirectory;
+            _nugetPackageDirectories = nugetPackageDirectories;
         }
 
-        private static string GetDefaultPackageDirectory(IEnvironment environment) => 
-            GetDefaultPackageDirectory(RuntimeEnvironment.OperatingSystemPlatform, environment);
+        private static string[] GetDefaultProbeDirectories(IEnvironment environment) =>
+            GetDefaultProbeDirectories(RuntimeEnvironment.OperatingSystemPlatform, environment);
 
-        internal static string GetDefaultPackageDirectory(Platform osPlatform, IEnvironment environment)
+        internal static string[] GetDefaultProbeDirectories(Platform osPlatform, IEnvironment environment)
         {
-            var packageDirectory = environment.GetEnvironmentVariable("NUGET_PACKAGES");
+#if !NETSTANDARD1_3            
+#if NETSTANDARD1_6
+            var probeDirectories = AppContext.GetData("PROBING_DIRECTORIES");
+#else
+            var probeDirectories = AppDomain.CurrentDomain.GetData("PROBING_DIRECTORIES");
+#endif
 
-            if (!string.IsNullOrEmpty(packageDirectory))
-            {
-                return packageDirectory;
-            }
+           var listOfDirectories = probeDirectories as string;
+
+           if (!string.IsNullOrEmpty(listOfDirectories))
+           {
+               return listOfDirectories.Split(new char [] {Path.PathSeparator}, StringSplitOptions.RemoveEmptyEntries );
+           }
+#endif
+
+           var packageDirectory = environment.GetEnvironmentVariable("NUGET_PACKAGES");
+
+           if (!string.IsNullOrEmpty(packageDirectory))
+           {
+               return new string[] {packageDirectory};
+           }
+
+           string basePath;
```

Comment:
> So, it is well understood that this is a dependence on an implementation detail of nuget 

Yes.  DependencyModel can't have a hard dependency on NuGet, so it can't call it for this information.

But in general, the problem you are pointing out is why we are doing this change.  NuGet added more locations to where it knows how to find assets (the "fallback" folder). This change is allowing the DependencyModel knowledge of these new locations.